src/cryptsetup/cryptsetup-tpm2.c

```diff
@@ -187,15 +189,25 @@ int acquire_tpm2_key(
                                 policy_hash_size,
                                 srk_buf,
                                 srk_buf_size,
-                                ret_decrypted_key,
-                                ret_decrypted_key_size);
+                                salt ? &secret : ret_decrypted_key,
+                                salt ? &secret_size : ret_decrypted_key_size);
                 /* We get this error in case there is an authentication policy mismatch. This should
                  * not happen, but this avoids confusing behavior, just in case. */
                 if (IN_SET(r, -EPERM, -ENOLCK))
                         return r;
                 if (r < 0)
                         continue;
 
+                /* Append salted pin to unsealed secret if there is a salt */
```

Comment:
This breaks unsealing volumes that were previously sealed without this salt. The fact that this salt is used needs to be recorded in the json metadata when enrolling, and used on unsealing only if it is recorded.

Reply:
No, the salt variable is set to NULL in such a case.